File: solutions/fortran/isogram/1/isogram.f90
module isogram
  implicit none
contains

  function isIsogram(phrase) result(no_repeats)
    character(len=*), intent(in) :: phrase
    logical :: no_repeats
    logical, dimension(26) :: has_occured 
    character :: c 
    integer ::  lt_no
    integer :: i
    no_repeats = .true.
    has_occured = (/(.false., I=1,26 )/)
    do i=1,len(phrase)
      c = phrase(i:i)
      if ((c>='a').and.(c<='z')) then
        lt_no = ichar(c) - ichar('a') + 1
        no_repeats = (no_repeats.and.(.not.(has_occured(lt_no))))
        has_occured(lt_no) = .true.
      else if ((c>='A').and.(c<='Z')) then
        lt_no = ichar(c) - ichar('A') + 1
        no_repeats = (no_repeats.and.(.not.(has_occured(lt_no))))
        has_occured(lt_no) = .true.
      end if
    end do
   end function isIsogram

end module isogram

File: solutions/dart/isogram/1/lib/isogram.dart
class Isogram {
  bool isIsogram(String phrase){
   var alphabet = Map<String, int>();
   bool result = true;
   var symbols = phrase.toLowerCase().split("");
   for(var s in symbols ){
     if(s=='-' || s==' ') continue;
     if(alphabet.containsKey(s)){
       result = false;
       break;
     }
     else{
       alphabet[s] = 1;
     }
   }
   return result;
  }
}
